src/main/java/com/revlearn/team1/dto/transaction/TransactionRequestDTO.java

```diff
@@ -1,7 +1,10 @@
 package com.revlearn.team1.dto.transaction;
+
+import com.revlearn.team1.model.TransactionModel;
+
+import java.util.List;
+
 public record TransactionRequestDTO(
-        Long toUserId,
-        Long fromUserId,
-        float price,
-        String description) {
-}
+        long id, // user id
+        TransactionModel transactionItem
```

Comment:
why do you need to pass in a TransactionModel here? When you make a request to create a transaction, aren't you only passing in the user_id, the description of the course or product that they're buying and then the price? (user_id, description, price) IDK what the TransactionModel is doing here.

Also everytime the transactionModel changes (add or delete a field), you're going to have to change the API request (which is why I'm guessing that DataInitializer is failing). If you could change this to be an ID or a reference, instead of an ACTUAL object, that would make things easier

Reply:
I was doing the TransactionModel so that we could store it for our own records in the database. I can remove it if we want

Reply:
I think we can create a Transaction in the database during this process! (I think that's what you're trying to do with Kafka, correct me if I'm wrong?) I just don't think all of that data needs to be passed in from the frontend. I think user_id, description, price, would be a super simple Request Body for the frontend-team to use but we can discuss that more if you disagree!

Reply:
That's a fair point. I'll reduce the DTO then.